humility-core/src/core.rs

```diff
@@ -199,7 +202,7 @@ impl ProbeCore {
             serial_number,
             unhalted_reads,
             halted: 0,
-            unhalted_read: crate::arch::unhalted_read_regions(),
+            unhalted_read: crate::arch_arm::unhalted_read_regions(),
```

Comment:
There is overall unhalted read support for the whole memory space, which is flagged by the `unhalted_reads` bool.  Then there are specific regions that support unhalted reads regardless of overall support for unhalted reads.  These regions are represented by `unhalted_read`.  On riscv, there are no regions which always support unhalted reads, so this is an arm specific option.

let me know if that makes sense, the fact the same name is used is confusing.

Additionally, probe-rs it doesn't seem to actually care if the board is explicitly halted when you try to read.

Reply:
Ah I see. That kind of makes sense. Does RV support ProbeCore at all? Or is that only ARM? I asked because this call to unhalted_read_regions seems to be done unconditionally here.

Reply:
It does support it.  With probe-rs, it will halt the core to do the read and restart it if necessary (As far as I can tell...?).   This means you don't have to explicitly halt, so even if the region is incorrect, it still works.

Reply:
Ack.